### PR TITLE
yq@3 - installs v3 of yq for backwards compatability

### PR DESCRIPTION
**Pull request by mikefarah**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Install version 3 of yq. I've recently release a major (backwards incompatible) upgrade (v4) - this has broken users scripts where they install yq via brew. The purpose of this PR is to allow them to continue using brew and yq v3 until a convenient time for them to upgrade to the latest v4.

Ref: mikefarah/yq#626

### CODE REVIEW

**Comment by SMillerDev** [comment]
Please indicate how this would fill our criteria for versioned formulas (https://docs.brew.sh/Versions)?

**Comment by mikefarah** [comment]
Cool I didn't realise that existed, will update

**Comment by mikefarah** [comment]
Fixed - reran all the checks (build from source, audit --strict, test)

**Comment by SMillerDev** [comment]
How long do you plan to support v3?

**Comment by mikefarah** [comment]
Hmm I'm thinking around 6 months - it depends on community feedback.

File: Formula/yq@3.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    (buildpath/"src/github.com/mikefarah/yq").install buildpath.children
+
+    cd "src/github.com/mikefarah/yq" do
+      system "go", "build", "-o", bin/"yq"
+      prefix.install_metafiles
+
+      (bash_completion/"yq").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=bash")
+      (zsh_completion/"_yq").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=zsh")
+      (fish_completion/"yq.fish").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=fish")
+    end
```
Comment (bayandin) [inline Formula/yq@3.rb:23]: If version 3.4.1 supports go modules, we can simplify it as we done it for yq 4:

```suggestion
    system "go", "build", "-ldflags", "-s -w", *std_go_args

    (bash_completion/"yq").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=bash")
    (zsh_completion/"_yq").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=zsh")
    (fish_completion/"yq.fish").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=fish")
```

Comment (mikefarah) [reply Formula/yq@3.rb:23]: The build works fine, but none of the completion commands work.

Although come to think of it - are they supposed to be included in a 'keg-only' version of yq?

Comment (SMillerDev) [reply Formula/yq@3.rb:23]: They are not afaik

Comment (mikefarah) [reply Formula/yq@3.rb]: Ok I've simplified the install, I did have to specify `yq` as the go build output to build the `yq` binary.

All the brew tests worked (build from source, test, audit strict)

**Comment by SMillerDev** [comment]
If you plan on deprecating this version in 6 months, please add `deprecated!` with a date.

**Comment by mikefarah** [comment]
Done

**Review by SMillerDev — APPROVED** [review]
Thanks @mikefarah  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**